src/ast.rs

```diff
@@ -106,9 +108,18 @@ impl<F, TraceArgs> Circuit<F, TraceArgs> {
         signal
     }
 
-    pub fn expose(&mut self, forward_signal: ForwardSignal) {
-        self.exposed.push(forward_signal);
+    pub fn expose(&mut self, signal: Queriable<F>, offset: ExposeOffset) {
+        match signal {
+            Queriable::Forward(..) | Queriable::Shared(..) => {
+                self.exposed.push((signal, offset));
+            }
+            _ => {
+                // TODO: Add a proper error here
+                panic!("cannot expose fixed or internal signals")
+            }
+        }
     }
+    
```

Comment:
Checking for Forward and Shared should already be done in the dsl `expose` function, so here we can simply push the `(signal, offset)`.

Reply:
I think it is better to do it in AST, if we are going to do it only in one place. As if someone modifies the AST without the DSL the check should be done